### DISCUSSION TITLE
Type error: Cannot find module '../portfolio/node_modules/next-sanity' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by Nelis93** [vercel/next.js#52662] [Help]
### Summary

Hi, first time trying to deploy with vercel. I get the following error:

Type error: Cannot find module '../portfolio/node_modules/next-sanity' or its corresponding type declarations.
import { createClient } from "../portfolio/node_modules/next-sanity";

Modules are included in .gitignore file, so I don't quite understand how this works. The reference works without issue in VScode. Next-sanity is installed properly and the environment variables are set. Would really appreciate any kind of help.

**Comment by icyJoseph** [comment]
Hi,

I think it is a good idea to move `components`, `utils`, `sanity.ts`,  inside `src`.

And, you are doing this, inside `sanity.ts`:

```ts
import { createClient } from "../portfolio/node_modules/next-sanity";
import createImageUrlBuilder from "../portfolio/node_modules/@sanity/image-url";
```

Which means, you go out from your project, and assume, return into your project, to the node_modules. 

However, when you deploy your repo is cloned and assigned a different project name, when going out from from `sanity.ts`, `../`, you are not guaranteed to find a `portfolio` directory. 

Just point to `./node_modules/next-sanity`, but rather move everything to `src`, and then just `import { createClient } from 'next-sanity';`

**Reply by Nelis93** [reply]
Thanks a lot for this!

**Reply by icyJoseph** [reply]
How did it go? I guess it still crashed out?

**Reply by Nelis93** [reply]
I'm struggling to reference the node modules in a way that works. Sanity has its own set of node modules, which are included in the .gitignore file. What's giving me issues now are the imports in the sanity.config.ts file. Is there perhaps a way to merge the sanity node modules into the general node modules directory? I'm just guessing here really. I've given you access to the repo so you can have a look (totally up to you of course).
